### DISCUSSION TITLE
jvm and native build with gitlab ci

### DISCUSSION THREAD

**Discussion by hamburml** [quarkusio/quarkus#33736] [Q&A]
Hello Devs,

https://gitlab.com/Hamburml/quarkus-dynamodb-example here is a blank quarkus example (i want to add dynamodb stuff, hence the name). But before I start I want two working stages, build-jvm and build-native and cache downloaded dependencies and docker images.

The build-jvm stage is working but I fail miserably at the build-native stage. At first I tried it with the quarkus-container-image-docker extension because I would prefer maven:latest as the base-image of the stage. This didn't work, the quarkus extension always failed because it doesn't find docker (but docker works, the Tests for example download docker images and run them).

I switched the image of the build-native stage to quay.io/quarkus/ubi-quarkus-mandrel-builder-image:22.3-java17 and use the maven-wrapper but now I see a maven-clean-plugin FileNotFound Exception.

Does anyone have an idea? Thanks for helping :)

PS: I would prefer the maven:latest base image if possible. I hope you have any ideas :)

[a comment by an automated account is not shown]

**Comment by hamburml** [comment]
Looks like I've made some progress, but I've run into another problem. If I set QUARKUS_NATIVE_CONTAINER_BUILD and QUARKUS_NATIVE_CONTAINER_RUNTIME I now see a `failed to pull builder error` and a `cannot run program docker` exception.
But some lines above the exception you see docker image pulls and localstack running.
https://gitlab.com/Hamburml/quarkus-dynamodb-example/-/jobs/4387069456

**Reply by bulldog98** [reply]
Most likely gitlab disallows nested containers, since it can be a security problem
